tahoe_lti/processors.py

```diff
@@ -88,17 +87,19 @@ def team_info(xblock):
     """
     from django.conf import settings
     features = getattr(settings, 'FEATURES', {})
-
-    if not features.get('ENABLE_TEAMS'):
+    user = get_xblock_user(xblock)
+    is_studio = getattr(xblock.runtime, 'is_author_mode', False)
+
+    if not (
+            features.get('ENABLE_TEAMS')  # Ensure the feature is enabled.
+            and user  # User should be logged in
+            and not is_studio  # Ensure we're in the LMS.
+    ):
```

Comment:
The big conditional gets hard to read. Maybe just three independent guard clauses instead?

```
if not features.get('ENABLE_TEAMS'); return  # Ensure the feature is enabled.
if not user: return  # User should be logged in
if is_studio: return  # Ensure we're in the LMS.
```

Obviously a bit of a style question, so I'm curious what others feel is most readable and will be easiest to extend/modify in the future.

Reply:
I've refactored it back to `if` statements. I don't have a strong preference to either.